SConstruct

```diff
@@ -531,141 +535,176 @@ class ObjectBuilder(object):
             if kwds:
                 env = env.Clone()
                 env.Prepend(**kwds)
-            path = UNITY_BUILD_DIRECTORY + filename
-            o = env.Object(Beast.variantFile(path, self.variant_dirs))
+            o = env.Object(Beast.variantFile(filename, self.variant_dirs))
             self.objects.append(o)
 
+def list_sources(base, suffixes):
+    def _iter(base):
+        for parent, dirs, files in os.walk(base):
+            files = [f for f in files if not f[0] == '.']
```

Comment:
I'd consider gluing this statement into the `for` statement below: either:

`for f in (f for f in files if not f[0] == '.'):`
or even just:

    for f in files:
        if not f.startswith('.'):

The reason is that you don't have to create that extra list.  Also, we generally prefer `startswith` for this because it works if the string is empty - though in this case you're assured that the string isn't empty.